content/contribute/toBabylon/HowToContribute.md

```diff
@@ -958,68 +958,70 @@ This will run all tests that have the word "material" in their name.
+#### Adding a new test
 
-```javascript
-await jestPuppeteer.debug();
-```
+To add a new test to the visualization tests, add the test data to this file: https://github.com/BabylonJS/Babylon.js/blob/master/packages/tools/tests/test/visualization/config.json
 
-anywhere in the test (in the node context, and not inside an evaluate function).
+The best way to do that is first create a playground, save it (using a local version of the playground, for example) and then add it at the end of the config file:
 
-Executing this will halt the test (taking the timeout into account!) and will allow you to look in the dev tools of the opened browser and debug the scene there.
+```json
+//...,
+{
+  "title": "My new visualization test", // the title of the test
+  "playgroundId": "#ABCDEF#12", // the playground snippet id
+  "referenceImage": "nameOfReferenceImage.png", // a name for the image snapshot that will be generated. Optional!
+  "excludedEngines": ["webgl1"], // optional lists of engine you don't want the test to run on
+  "renderCount": 2, // optional, the number of frames the scene should render. Default is 1. Good for animated scenes.
+},
+//...
+```
 
-The tests are set to automatically show any error received from the browser's console (doesn't work in Firefox). You will see the error in the jest output screen.
+To generate the snapshot, run the test once. The simplest would be to open the UI and run the test. The snapshot will be generated in the `tests/test/visualization/snapshots` folder. Make sure to add the snapshot to the repository. See the next section to know how to run a single test.
 
-For the time being the tests are defined in the visualization file itself, but this is temporary until the development is finalized.
+#### Running specific tests
 
-In the future it is recommended to have visualization tests in the dev packages themselves and not in an external package. for the time being, the visualization tests are located in @tools/visualization-tests
+The recommended way to run a specific test is to use the UI. To do that run
 
-#### Running specific tests
+```shell
+npm run test:visualization:ui
+```
 
-To run a specific test on a specific engine, use:
+This will show you the tests for all 3 engines and you will be able to run them one by one.
+
+If you want to display only one specific engine add the following:
 
 ```shell
-npm run test:visualization -- -i "engineName" -t "test name"
+npm run test:visualization:ui -- --project="webgl2"
 ```
 
-For example:
+This will only show the tests for the webgl2 engine.
+
+Note that when making changes to the list of tests the UI will reload to show it. So you can update the JSON while the UI is loaded without the need to run it again.
+
+If you don't want to use the UI you can run an individual test by providing its title:
 
 ```shell
-npm run test:visualization -- -i "webgl2" -t "Particle subemitters"
+npm run test:visualization -- --project="webgl2" -g "Title Of The Test"

```

Comment:
Just to be sure: can you confirm it is now "-g" instead of "-t"?

Reply:
yes, playwright uses `-g`, and without `=`. Confusing? YES!....